plugin/src/main/java/io/spine/tools/gradle/bootstrap/ConfigurationSensitive.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.gradle.bootstrap;
+
+/**
+ * A configuration element which may enforce particular dependencies necessary for their work.
+ *
+ * <p>In Spine Bootstrap plugin, for some elements, it's necessary to have the particular
+ * dependency no lower than version {@code X} in the project.
+ *
+ * <p>Such elements are represented by this interface and may force the necessary project
+ * configuration to make sure the correct dependency version is used, regardless of the other
+ * project environment.
+ */
+interface ConfigurationSensitive {
```

Comment:
This interface is used in Gradle extensions. The problem with that is that `forceConfiguration()` and `disableConfigurationEnforcement()` are now accessible to the end-users via the build script DSL.

Please consider hiding them and making the `forceConfiguration` field the only way to configure this.